### DISCUSSION TITLE
Force time intervals to appear

### DISCUSSION THREAD

**Discussion by HuifangYeo** [atoti/atoti#726] [Q&A] [answered]
I have transactions coming in at random timing e.g. 9am, 11am.
I would like to display the full time scale e.g. 00hr, 01hr, 02hr... 22hr, 23hr.

For this, I have created the Time table which consists of all the hours, with its minutes and seconds. I could only get the full timescale if I use the Time table as my base store. Is there anyway I could use my Payment table as the base store instead?
![image](https://user-images.githubusercontent.com/35210495/206147946-d916f2f5-f4fa-45a6-ba6c-c6b55052c2d4.png)

**Comment by HuifangYeo** [comment]
I set my payment table to be the base and perform a join to the Time table using only Time key as shown below:
![image](https://user-images.githubusercontent.com/35210495/206372432-cbd07184-d56f-4127-b302-88976d04b139.png)

I am able to get the time interval displayed when I remove `non empty` from the MDX:
![image](https://user-images.githubusercontent.com/35210495/206372658-d594d331-59dd-45e9-ad7e-597ba1a16d3b.png)

Question:
Removing `non empty` works when add in the levels (`l["Hour"]`, `l["Minute"]`, `l["Second"]`) manually.

If I create date hierarchy as shown below, the factless hierarchy no longer works.
```
cube.create_date_hierarchy(
    "Dates",
    column=payment_tbl["Settlement_Date"],
    levels={"Year": "yyyy", "Quarter": "QQQ", "Month": "MM", "Day": "dd"},
)

h["Dates"] = {
    **h["Dates"].levels,
    "Hour": l["Hour"],
    "Minute": l["Minute"],
    "Second": l["Second"],
}
``` 
Is there anyway I keep the date hierarchy and still display the factless hierarchy?

**Reply by tibdex** [reply]
If you keep the hierarchies `Dates` and the one coming from the `Time` table separate (i.e. not merge them as seen in the last part of your example), you can still see all the times.

**Comment by HuifangYeo** [comment, marked as answer]
Thanks @tibdex 

Indeed, so I joined only by the `Settlement_time`:
![image](https://user-images.githubusercontent.com/35210495/206958122-65cf8edc-35f1-4733-bfd1-4feccb403ca8.png)

and I created 2 multilevel hierarchies:

```
cube.create_date_hierarchy(
    "Dates",
    column=payment_tbl["Settlement_Date"],
    levels={"Year": "yyyy", "Quarter": "QQQ", "Month": "MM", "Day": "dd"},
)

h["Dates"] = {**h["Dates"].levels, "Time": payment_tbl["Settlement_Time"]}

h["Time"] = [l["Hour"], l["Minute"], l["Second"]]
```

By remove `Non Empty` from my MDX query, I'm able to see those factless levels.
